src/figuresClasses.ts

```diff
@@ -2,10 +1,115 @@
+export interface Figure {
+  color: string;
+  a?: number;
+  b?: number;
+  c?: number;
+  radius?: number;
+  width?: number;
+  height?: number;
+}
+
+export type FiguresData = {
+  shape: string;
+  color: string;
+  area: number;
+};
+
+export class Triangle implements Figure {
+  public shape = 'triangle';
+
+  public area = this.getArea();
+
+  constructor(
+    public color: string,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if (
+      color === undefined ||
+      a === undefined ||
+      b === undefined ||
+      c === undefined
+    ) {
+      throw new Error('All values must be entered');
+    }
+
+    if (a <= 0 || b <= 0 || c <= 0) {
+      throw new Error('All sides values must be > 0');
```

Comment:
[CODE STYLE] - The error message should be more descriptive. Consider 'Triangle side lengths must be greater than zero.'